providers/azureblob/src/main/java/org/jclouds/azure/storage/filters/SharedKeyLiteAuthentication.java

```diff
@@ -85,23 +91,89 @@ public SharedKeyLiteAuthentication(SignatureWire signatureWire,
       this.crypto = crypto;
       this.utils = utils;
       this.signatureWire = signatureWire;
+	  this.storageUrl = URI.create("https://" + creds.get().identity + ".blob.core.windows.net/");
       this.creds = creds;
+	  this.identity = creds.get().identity;
+	  this.credential = creds.get().credential;
       this.timeStampProvider = timeStampProvider;
+	  this.sasAuthentication = false;
    }
-
+   
+   /** 
+   * this is an updated filter method, which decides whether the SAS or SharedKeyLite 
+   * is used and applies the right filtering.  
+   * 
+   */
    public HttpRequest filter(HttpRequest request) throws HttpException {
+      this.sasAuthentication = authSAS(this.credential);
```

Comment:
This could be set in the constructor instead of checking this on every request.

Reply:
Done!